.github/scripts/format-check

```diff
@@ -1,14 +1,12 @@
-#!/bin/bash -eux
+#!/bin/bash -ae
 
-unformatted() { 
-    set -x
-    make format
-    test -z "$(git status --porcelain)"
+unformatted() {
+  set -x
+  make format
+  test -z "$(git status --porcelain)"
 }
 
-export -f unformatted
-
-source conda/etc/profile.d/conda.sh
+source /tmp/conda/etc/profile.d/conda.sh
```

Comment:
See the `install-conda` script below, but: Previously our CI jobs were downloading the Miniforge installer and installing conda to the root of the git clone directory. We cannot do that with this format check, because `git status --porcelain` will report the unversioned files, while will make it non-silent as we'd like. So, download Miniforge and install conda to `/tmp/conda`. Other CI scripts are updated to reflect this change, too.